### ISSUE TITLE
Urllib3 import error on pyelasticsearch import

### ISSUE THREAD

**Issue by delvv** [pyelasticsearch/pyelasticsearch#174]
After some sort of Ubuntu update that hit last night (running Ubuntu 14.04.2 LTS), we were unable to import pyelasticsearch, due to an incompatibility with the 1.10.4 version of urllib3.  Note that this is essentially identical to the import error reported here: https://github.com/pyelasticsearch/pyelasticsearch/issues/165

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/pyelasticsearch/__init__.py", line 3, in <module>
    from pyelasticsearch.client import ElasticSearch
  File "/usr/local/lib/python2.7/dist-packages/pyelasticsearch/client.py", line 18, in <module>
    from pyelasticsearch.exceptions import (ElasticHttpError,
  File "/usr/local/lib/python2.7/dist-packages/pyelasticsearch/exceptions.py", line 1, in <module>
    from urllib3.exceptions import TimeoutError as Timeout, ConnectionError
ImportError: cannot import name ConnectionError
```

Running pyelasticsearch 1.2.3:

```
$ pip freeze | grep pyelasticsearch
pyelasticsearch==1.2.3
```

python and urllib3 versions:
```
$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib3
>>> urllib3.__version__
'1.10.4'
>>> urllib3.__file__
'/usr/lib/python2.7/dist-packages/urllib3/__init__.pyc'
```

After deleting the copy of urllib in the /usr/lib/python2.7/dist-packages and forcing pip to update the /usr/local/lib/python2.7/dist-packages to urllib-1.8.3, everything started working fine again.

**Comment by erikrose** [comment]
Hmm. The copy of urllib3 I just downloaded has the following in its `exceptions` module:

```python
#: Renamed to ProtocolError but aliased for backwards compatibility.
ConnectionError = ProtocolError
```
Does that occur in Ubuntu's?

**Comment by erikrose** [comment]
No, it doesn't. In fact, the Ubuntu version bears little resemblance to the upstream. How weird!

**Comment by erikrose** [comment]
Not only is the ConnectionError alias missing, but ProtocolError is missing as well. And the calling code in connectionpool.py (for one) is consequently a lot different.

**Comment by erikrose** [comment]
Aha. I don't think you upgraded to 1.10.4 after all. Ubuntu 14 [has only urllib3 1.7.1 packaged](http://packages.ubuntu.com/search?keywords=python-urllib3). Does your `aptitude show python-urllib3` show 1.7.1? Mine does:

```
$ aptitude show python-urllib3
Package: python-urllib3           
State: installed
Automatically installed: no
Version: 1.7.1-1ubuntu3
```

pyelasticsearch currently requires urllib3 1.8 or better. FWIW, I suggest using virtualenvs to keep clear of old system packages. To alleviate trust issues with installing things from PyPI, I use [peep](https://pypi.python.org/pypi/peep/).

**Comment by delvv** [comment]
Huh, you are absolutely correct, I looked at aptitude as well as the files I backed up from the Ubuntu install of urllib3 and they are all version 1.7.1 and look like they are unchanged - so that wasn't the source of the "bad" urllib3 import at all.  Will have to look into this more - there was clearly a bad package that snuck in somewhere (i.e. pip), I thought it was the one from Ubuntu but clearly not.
